modules/nlu/src/backend/application/scoped/prediction-handler.ts

```diff
@@ -67,21 +67,21 @@ export class ScopedPredictionHandler {
   }
 
   private async tryPredictInLanguage(textInput: string, language: string): Promise<RawEventUnderstanding | undefined> {
-    if (!this._modelsByLang[language]) {
+    const modelId = this._modelsByLang[language]
+    if (!modelId) {
       return
     }
 
     try {
-      const rawOriginalOutput = await this._engine.predict(this._botId, textInput, this._modelsByLang[language])
+      const rawOriginalOutput = await this._engine.predict(this._botId, textInput, modelId)
       const originalOutput = mapPredictOutput(rawOriginalOutput)
       const { spellChecked } = originalOutput
-      return { ...originalOutput, spellChecked, errored: false, language }
+      return { ...originalOutput, spellChecked, errored: false, language, modelId }
     } catch (err) {
-      const modelId = this._modelsByLang[language]
       const msg = `An error occured when predicting for input "${textInput}" with model ${modelId}`
       this._logger.attachError(err).error(msg)
 
-      return { errored: true, language }
+      return { errored: true, language, modelId: undefined }
```

Comment:
@franklevasseur Can you check this. Not sure what this does

Reply:
this makes sense, its to check in the frontend that the bot was trained and has a usable model.

@samuelmasse this should go in `event.nlu.modelId`